### PR TITLE
[ENHANCEMENT] Allow dynamic selection of authentication provider in percli

### PR DESCRIPTION
**Pull request by celian-garcia**
# Description

This PR clarifies the current support of different authentication providers by ``percli``. It removes any doubt with clear messages on what we can do and cannot and it starts the implementation with a simple selector.

# Screenshots

### When backend support several auth provider. The command line ask you to select
image

### ... with an error message for the moment for OIDC/OAuth 2.0, as it is not supported by the CLI
image

### When the backend does not support native, we cannot set username or password arguments
image
image

### One can set the provider through argument (it might be useful when we'll implement the client credentials flow)
image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: internal/cli/cmd/login/login.go [hunk 1]
```diff
@@ -128,24 +124,113 @@ func (o *option) authAndSetToken() error {
 	return nil
 }
 
-func (o *option) readAndSetCredentialInput() error {
-	if len(o.username) == 0 {
-		_, _ = fmt.Fprint(o.writer, "Username: ")
-		user, err := read.FromStdin()
-		if err != nil {
-			return err
+type MyWriteCloser struct {
```
Comment (celian-garcia) [inline internal/cli/cmd/login/login.go:127]: This is not ideal. 
There is an issue about that in promptui...
https://github.com/manifoldco/promptui/issues/211

Now reviewing the project and the lack of maintenance. I probably need to change my lib...

File: internal/cli/cmd/login/login.go [hunk 2]
```diff
@@ -128,24 +124,113 @@ func (o *option) authAndSetToken() error {
...
-	if len(o.password) == 0 {
-		_, _ = fmt.Fprint(o.writer, "Password: ")
-		password, err := readPassword()
-		if err != nil {
-			return err
+		if len(o.password) == 0 {
+			prompt := promptui.Prompt{
+				Label:  "Password",
+				Stdout: MyWriteCloser{writer: o.writer},
+				Mask:   ' ',
```
Comment (celian-garcia) [inline internal/cli/cmd/login/login.go:157]: I remember that this was unwanted by @Nexucis: we see the number of character of the password
image

I will see if I can find a lib that really hides it or get back the readpassword function

**Review by celian-garcia — COMMENTED** [review]
I answer myself here. More or less the idea is here. Just the library needs to be changed

File: internal/cli/cmd/login/login.go [hunk 3]
```diff
@@ -14,46 +14,42 @@
...
-	refreshToken string
-	insecureTLS  bool
-	apiClient    api.ClientInterface
-	restConfig   perseshttp.RestConfigClient
+	writer           io.Writer
+	url              string
+	username         string
+	password         string
+	loginKind        loginKind
+	selectedProvider string
```
Comment (celian-garcia) [inline internal/cli/cmd/login/login.go:47]: for non interactive use cases, we want to be able to select the provider from an argument

File: internal/cli/cmd/login/login.go (hunk 1 above)
Comment (celian-garcia) [reply internal/cli/cmd/login/login.go]: used huh instead

File: internal/cli/cmd/login/login.go (hunk 2 above)
Comment (celian-garcia) [reply internal/cli/cmd/login/login.go]: Used huh instead, that should support it soon https://github.com/charmbracelet/huh/issues/129

File: internal/cli/cmd/login/login.go (hunk 3 above)
Comment (celian-garcia) [reply internal/cli/cmd/login/login.go]: done

**Review by Gladorme — APPROVED** [review]

**Pull request merged**